internal/database/drivers/mongodb/mongodb.go

```diff
@@ -154,25 +154,35 @@ func (m *Database) collections(ctx context.Context, filter bson.D) ([]string, er
 }
 
 // GetEvents gets all events information.
-func (m *Database) GetEvents(ctx context.Context, request requests.MultipleEventsRequest) ([]drivers.EiffelEvent, error) {
+func (m *Database) GetEvents(ctx context.Context, request requests.MultipleEventsRequest) ([]drivers.EiffelEvent, int64, error) {
 	filter, err := buildFilter(request.Conditions)
 	if err != nil {
 		m.logger.Errorf("Database: %v", err)
-		return nil, err
+		return nil, 0, err
 	}
 	collections, err := m.collections(ctx, filter)
 	if err != nil {
 		m.logger.Errorf("Database: %v", err)
-		return nil, err
+		return nil, 0, err
 	}
 
 	m.logger.Debugf("fetching events from %d collections", len(collections))
 	var allEvents []drivers.EiffelEvent
+	var numberOfDocuments int64
 	for _, collection := range collections {
 		var events []drivers.EiffelEvent
-		cursor, err := m.database.Collection(collection).Find(ctx, filter,
-			// Remove the _id field from the resulting document.
-			options.Find().SetProjection(bson.M{"_id": 0}))
+		findOptions := options.Find()
+		// Remove the _id field from the resulting document.
+		findOptions.SetProjection(bson.M{"_id": 0})
+		// PageNo starts at 1 instead of 0 so we need to decrement it by 1 here.
+		findOptions.SetSkip(int64((request.PageNo - 1) * request.PageSize))
+		findOptions.SetLimit(int64(request.PageSize))
+
+		col := m.database.Collection(collection)
+		count, _ := col.CountDocuments(ctx, filter, &options.CountOptions{})
```

Comment:
Ouch. Wouldn't this introduce a significant increase in the query cost? It's cheaper to count documents than to return them, but still. Could we avoid this in certain cases, like if the number of returned documents is less than request.PageSize and request.PageNo == 1?

Reply:
Done